instrumentation-core/spring/README.md

```diff
@@ -340,6 +341,324 @@ After running Jaeger locally, navigate to the url below. Make sure to refresh th
+public class SecondServiceController {
+   @Autowired
+   private Tracer tracer;
+
+   @GetMapping
+   public String secondTracedMethod() {
+      return "It's time to get a watch";
+   }
+}
+```
+
+#### Create Controller Interceptor
+
+Add the class below to wrap all requests to the SecondServiceController in a span. This class will call the preHandle method before the RestController is entered and the postHandle method after a response is created. 
+
+The preHandle method starts a span for each request. The postHandle method closes the span and adds the span context to the response header. This implementation is shown below:    
+
+```java
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+import io.grpc.Context;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import io.opentelemetry.trace.TracingContextUtils;
+
+@Component
+public class ControllerTraceInterceptor implements HandlerInterceptor {
+
+   private static final HttpTextFormat.Getter<HttpServletRequest> getter =
+         new HttpTextFormat.Getter<HttpServletRequest>() {
+            public String get(HttpServletRequest req, String key) {
+               return req.getHeader(key);
+            }
+         };
+
+   private static final HttpTextFormat.Setter<HttpServletResponse> setter =
+         new HttpTextFormat.Setter<HttpServletResponse>() {
+            @Override
+            public void set(HttpServletResponse response, String key, String value) {
+               response.addHeader(key, value);
+            }
+         };
+
+   @Autowired
+   private Tracer tracer;
+
+   @Override
+   public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+         throws Exception {
+      Context context = OpenTelemetry.getPropagators().getHttpTextFormat()
+          .extract(Context.current(), request, getter);
+      Span span = createSpanWithParent(request, context);
+      span.setAttribute("handler", "pre");
+      tracer.withSpan(span);
+
+      return true;
+   }
+
+   @Override
+   public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
+         ModelAndView modelAndView) throws Exception {
+
+      Span currentSpan = tracer.getCurrentSpan();
+      currentSpan.setAttribute("handler", "post");
+      OpenTelemetry.getPropagators().getHttpTextFormat().inject(Context.current(), response, setter);
+      currentSpan.end();
+   }
+
+   @Override
+   public void afterCompletion(HttpServletRequest request, HttpServletResponse response,
+         Object handler, Exception exception) throws Exception {}
```

Comment:
would it be better to end the span in `afterCompletion`?

Reply:
bumping this, it seems github is hiding a couple of conversations even though they aren't marked resolved